core/src/test/java/org/apache/iceberg/TestManifestReader.java

```diff
@@ -44,7 +44,8 @@ public class TestManifestReader extends TestBase {
               "fileOrdinal",
               "fileSequenceNumber",
               "fromProjectionPos",
-              "manifestLocation")
+              "manifestLocation",
+              "partitionData.partitionType.fieldsById")
```

Comment:
What did we need this?

Reply:
When the tests were moved to JUnit 5, the previous assertions were updated to use reflection and a recursive equality check. This lazy field is now present in the value that gets returned, although it doesn't really affect anything. This change gets the tests to work and I don't want to fix the real issue -- that the switch to JUnit5 wasn't done right and checks more than the original tests did rather than porting the original assertions.

Needless to say, I'm not a fan of how that conversion went, but it isn't something we should fix here.